src/cloud-api-adaptor/test/e2e/common_suite.go

```diff
@@ -615,3 +615,46 @@ func DoTestKbsKeyReleaseForFailure(t *testing.T, e env.Environment, assert Cloud
 
 	NewTestCase(t, e, "DoTestKbsKeyReleaseForFailure", assert, "Kbs key release is failed").WithPod(pod).WithTestCommands(testCommands).Run()
 }
+
+func DoTestRestrictivePolicyBlocksExec(t *testing.T, e env.Environment, assert CloudAssert) {
+	allowAllExceptExecPolicyFilePath := "../../podvm/files/etc/kata-opa/allow-all-except-exec-process.rego"
+	podName := "policy-exec-rejected"
+	pod := NewPodWithPolicy(E2eNamespace, podName, allowAllExceptExecPolicyFilePath)
+
+	testCommands := []TestCommand{
+		{
+			Command:       []string{"ls"},
+			ContainerName: pod.Spec.Containers[0].Name,
+			TestErrorFn: func(err error) bool {
+				if strings.Contains(err.Error(), "failed to exec in container") && strings.Contains(err.Error(), "ExecProcessRequest is blocked by policy") {
+					log.Infof("Exec process was blocked %s", err.Error())
```

Comment:
Hi @stevenhorsman !

I didn't see that Infof message in the job's logs: https://productionresultssa11.blob.core.windows.net/actions-results/d4864003-7eda-43f2-b10f-794a54b12c7c/workflow-job-run-a0fa4af4-13b0-56fc-66c4-e46ff528b023/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-07-03T14%3A00%3A25Z&sig=I4u%2BC3Y20%2FV7Jjz3CRWRXhXd5VCpEvJs%2BdU7JaIZZys%3D&ske=2024-07-04T01%3A14%3A58Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-07-03T13%3A14%3A58Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2023-11-03&sp=r&spr=https&sr=b&st=2024-07-03T13%3A50%3A20Z&sv=2023-11-03

should I?

Reply:
Due to the kata payload not building for a couple of week the required kata fix I added isn't in our operator install, so we need to fix that before we can probably test it here. Saying that it suggests there is a false positive here, so I need to dig in here more. Thanks for the prompt!

Reply:
I've potentially founds a but in the code that test commands against the pod, that if the pod isn't found it just succeeds, so debugging that now.

Reply:
Well it's failing now, but not it the way I'd expect. I'll need to dig into this locally to debug it...

Reply:
Ok, found the problem - we didn't have anything to handle the case that we expected an error, but didn't find one, so I've hopefully plugged that gap and the libvirt tests should fail now (until we get a new kata-payload with the policy support fix)

Reply:
It's working now in https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/9807929350/job/27084196583?pr=1876
```
assessment_runner.go:417: We expected an error from Pod policy-exec-rejected, but it was not found
```